pkg/library/command.go

```diff
@@ -0,0 +1,89 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package library
+
+import (
+	"fmt"
+
+	"github.com/devfile/api/pkg/apis/workspaces/v1alpha1"
+)
+
+func getCommandType(command v1alpha1.Command) (v1alpha1.CommandType, error) {
+	err := command.Normalize()
+	if err != nil {
+		return "", err
+	}
+	return command.CommandType, nil
+}
+
+func getCommandsForIds(ids []string, commands []v1alpha1.Command) ([]v1alpha1.Command, error) {
+	var resolvedCommands []v1alpha1.Command
+
+	for _, id := range ids {
+		resolvedCommand, err := getCommandById(id, commands)
+		if err != nil {
+			return nil, err
+		}
+		resolvedCommands = append(resolvedCommands, *resolvedCommand)
+	}
+
+	return resolvedCommands, nil
+}
+
+func getCommandById(id string, commands []v1alpha1.Command) (*v1alpha1.Command, error) {
+	for _, command := range commands {
+		commandKey, err := command.Key()
+		if err != nil {
+			return nil, err
+		}
+		if commandKey == id {
+			return &command, nil
+		}
+	}
+	return nil, fmt.Errorf("no command with key %s is defined", id)
```

Comment:
In the future I assume that these rules should be implemented in the `validation` package of the `devfile/api` repo.
As soon as we have something working here, we should sync with other teams to move this into the common `validation` code.